### PR TITLE
Move timeline by one step when track with increment property is created

### LINKED ISSUE #33681: Animation editor timeline playhead does not advance when adding a key if the confirmation dialog appears
**Godot version:**
3.2 beta 1 mono

**OS/device including version:**
Win10 Home 64

**Issue description:**
It can be seen that when you enter the first frame by clicking on the key icon to add a keyframe, the frame is entered in the animation editor but the timeline did not move, then when you click on the key icon again , the second frame overlaps the first frame.
The first frame disappears, it cannot be selected.
Although you double clicked on the key icon, only the second frame was added.
The first frame was overwritten.

**_The problem was that the timeline did not shift when the first frame was introduced._**

I would like to add something, add a new animation by pressing the "Animation" -> "New" button should bring the timeline to 0.

**Steps to reproduce:**
Add the initial keyframe in the animation editor and then a new keyframe.

The first frame will be overwritten by the second frame, then even though you pressed the keyframe add button twice, only the second frame was added to the animation editor.

GIF

**Minimal reproduction project:**
[animationEditorBug.zip](https://github.com/godotengine/godot/files/3855960/animationEditorBug.zip)

### PR DESCRIPTION
**Pull request by rathren**
This closes #33681

### CODE REVIEW

**Comment by rathren** [comment]
I hope Cloning the last one and making a new pull request was the proper way to do this. If not please tell me :)

**Comment by akien-mga** [comment]
> I hope Cloning the last one and making a new pull request was the proper way to do this. If not please tell me :)

The recommeded workflow would have been to modify the pull request directly by force pushing changes to your remote branch. It's also recommended to create dedicated branches for pull requests instead of using your `master` branch, which would prevent you from making several PRs until the one based on the `master` branch is merged (otherwise any other branch based off your `master` branch will include those commits).

You should also squash the two commits in this PR into one.

See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) for a tutorial on all the above.

**Comment by rathren** [comment]
oh I see thank you. Should/can I still do the things you mentioned now or just do it like this in the future?

**Comment by akien-mga** [comment]
You should still squash the two commits, and do future edits via `git commit --amend` and force pushes to this branch/PR.

But for the rest indeed, you can do it next time (you can't change the base branch in this PR without closing it and opening yet another one).

**Comment by rathren** [comment]
ok I managed to squash it :D

**Comment by YeldhamDev** [comment]
Would be nice if the "Hack" comments could be a little more descriptive, like telling why the hack is there and what it's doing,

**Comment by rathren** [comment]
> Would be nice if the "Hack" comments could be a little more descriptive, like telling why the hack is there and what it's doing,

I agree but it is actually something already used by someone else little above and I am not 100% myself how it does it. But if someone has a good comment for it I can add it for both places it got used. :)

File: editor/animation_track_editor.cpp [hunk 1]
```diff
@@ -4066,6 +4066,29 @@ int AnimationTrackEditor::_confirm_insert(InsertData p_id, int p_last_track, boo
...
+		animation->track_set_path(animation->get_track_count() - 1, p_id.path);
+		PropertyInfo h = _find_hint_for_track(animation->get_track_count() - 1, np);
+		animation->remove_track(animation->get_track_count() - 1); //hack
+
+		if (h.usage & PROPERTY_USAGE_KEYING_INCREMENTS) {
+
+			// move timeline by one step
+			float step = animation->get_step();
+			if (step == 0)
+				step = 1;
```
Comment (aaronfranke) [inline editor/animation_track_editor.cpp:4082]: The current master branch has new style rules, including enforcing braces. Your PR will need to be rebased and updated to conform to these in order for it to be accepted. For more information, read this documentation article: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html

**Comment by KoBeWi** [comment]
So I just closed #33681 because it's apparently already fixed. If you can confirm, the PR can be closed.

**Comment by avencherus** [comment]
> So I just closed #33681 because it's apparently already fixed. If you can confirm, the PR can be closed.

Still an issue as of 83f7c2d957fa4abd8e78de63603a50972074114d from what I can tell.  Was the fix more recent, do you have a link to it?

In the build I have from that commit, clicking the inspector keyframe for a sprite frame property doesn't advance a step on the animation timeline for the first keyframe.  It does advance in frame number.  So the next time it is clicked it will move on to 1, but overwrite the first keyframe then advance to frame 2 while moving the timeline by the time increment.

Testing the PR, it appears to be functioning properly.

To see the broken behavior watch as the first keyframe is overwritten.  The end values are 1 and 2.  You'll see it advanced on the second click and begins to behave as it was intended from there on.

ezgif com-gif-maker

**Comment by KoBeWi** [comment]
@avencherus Thanks for checking this, seems like it only happens when the insert confirmation dialog appears. I reopened the issue.

**Comment by akien-mga** [comment]
Superseded by #48995. Thanks for the contribution nonetheless!

**Pull request closed without merging**